### PR TITLE
Update Kafka to 3.1

### LINKED ISSUE #5: Prepare Kafka 3.0 upgrade
Kafka 3.0 is already out for a while. We should try whether there are any problems when running on Kafka 3.0.

### PR DESCRIPTION
**Pull request by raminqaf**
Fixes #5 and updates deprecated functions and interfaces. Moreover, this PR updates other dependencies.

### CODE REVIEW

File: common/src/main/java/com/bakdata/quick/common/security/TokenPropagator.java [hunk 1]
```diff
@@ -36,12 +36,12 @@
 @Requires(property = SecurityConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
 @Filter(Filter.MATCH_ALL_PATTERN)
 @Slf4j
-public class TokenPropagator extends OncePerRequestHttpServerFilter {
+
```
Comment (torbsto) [inline common/src/main/java/com/bakdata/quick/common/security/TokenPropagator.java:39]: Remove line

File: common/src/main/java/com/bakdata/quick/common/security/TokenPropagator.java [hunk 2]
```diff
@@ -36,12 +36,12 @@
...
+public class TokenPropagator implements HttpServerFilter {
     private static final String API_KEY_HEADER_NAME = "X-API-Key";
 
     @Override
-    protected Publisher<MutableHttpResponse<?>> doFilterOnce(final HttpRequest<?> request,
-        final ServerFilterChain chain) {
+    public Publisher<MutableHttpResponse<?>> doFilter(HttpRequest<?> request, ServerFilterChain chain) {
```
Comment (torbsto) [inline common/src/main/java/com/bakdata/quick/common/security/TokenPropagator.java:44]: final arguments

File: mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java [hunk 3]
```diff
@@ -50,16 +54,20 @@ public StreamsStateController(final QueryContextProvider contextProvider) {
...
-        return Seq.seq(metadata.topicPartitions())
-            .map(partition ->
-                new PartitionAddress(partition.partition(), metadata.host(), metadata.port()));
+    private static Stream<PartitionAddress> getAddressesForPartitions(final StreamsMetadata metadata) {
+        return metadata.topicPartitions().stream()
+            .map(partition -> new PartitionAddress(partition.partition(), metadata.host(), metadata.port()));
+    }
+
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Set<Object> seen = ConcurrentHashMap.newKeySet();
```
Comment (torbsto) [inline mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java:69]:
```suggestion
        final Set<Object> seen = ConcurrentHashMap.newKeySet();
```

File: mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java [hunk 4]
```diff
@@ -50,16 +54,20 @@ public StreamsStateController(final QueryContextProvider contextProvider) {
...
-    private static Seq<PartitionAddress> getAddressesForPartitions(final StreamsMetadata metadata) {
-        return Seq.seq(metadata.topicPartitions())
-            .map(partition ->
-                new PartitionAddress(partition.partition(), metadata.host(), metadata.port()));
+    private static Stream<PartitionAddress> getAddressesForPartitions(final StreamsMetadata metadata) {
+        return metadata.topicPartitions().stream()
+            .map(partition -> new PartitionAddress(partition.partition(), metadata.host(), metadata.port()));
+    }
+
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
```
Comment (torbsto) [inline mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java:68]:
```suggestion
    private static <T> Predicate<T> distinctByKey(final Function<? super T, ?> keyExtractor) {
```

File: mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java [hunk 5]
```diff
@@ -50,16 +54,20 @@ public StreamsStateController(final QueryContextProvider contextProvider) {
...
+        Set<Object> seen = ConcurrentHashMap.newKeySet();
+        return t -> seen.add(keyExtractor.apply(t));
```
Comment (torbsto) [inline mirror/src/main/java/com/bakdata/quick/mirror/StreamsStateController.java:70]:
```suggestion
        return element -> seen.add(keyExtractor.apply(element));
```

File: common/src/main/java/com/bakdata/quick/common/graphql/GraphQLUtils.java [hunk 6]
```diff
@@ -60,12 +61,12 @@ private GraphQLUtils() {
      * Checks whether two TypeDefinitions are the same.
      *
      * <p>
-     * This is necessary as we consider equality when the fields are equals. {@link TypeDefinition#equals(Object)} also
```
Comment (torbsto) [inline common/src/main/java/com/bakdata/quick/common/graphql/GraphQLUtils.java:63]: Why did you remove this? Referencing `equals` is indented

**Review by torbsto — APPROVED** [review]

File: common/src/main/java/com/bakdata/quick/common/graphql/GraphQLUtils.java (hunk 6 above)
Comment (raminqaf) [reply common/src/main/java/com/bakdata/quick/common/graphql/GraphQLUtils.java:63]: My IntelliJ marked the `euqulas` red because it couldn't find it. But I can revert it!

Comment (torbsto) [reply common/src/main/java/com/bakdata/quick/common/graphql/GraphQLUtils.java:63]: Ah okay, it's okay then
